src/app/time-picker/time-picker.sample.html

```diff
@@ -72,5 +72,37 @@ <h4 class="sample-title">Templated Time Picker</h4>
                 </igx-time-picker>
             </div>
         </article>
+        <article class="timepicker-column">
+            <h4 class="sample-title">Templated Time Picker</h4>
+            <p class="sample-description">Time picker with dropdown mode</p>
+            <div class="preview">
+                <igx-time-picker [value]="today" format="HH:mm" mode="dropdown">
+                    <ng-template igxTimePickerTemplate let-openDialog="openDialog" let-value="value" let-displayTime="displayTime">
+                        <igx-input-group>
+                            <input #dropDownTarget igxInput class="date-picker" [(value)]="displayTime"/>
+                            <igx-suffix>
+                                <igx-icon class="date-picker-icon" (click)="openDialog()">access_alarm</igx-icon>
+                            </igx-suffix>
+                        </igx-input-group>
+                    </ng-template>
+                </igx-time-picker>
+            </div>
+        </article>
+        <article class="timepicker-column">
+            <h4 class="sample-title">Templated Time Picker</h4>
+            <p class="sample-description">Time picker with dropdown mode + overlay settings </p>
+            <div class="preview">
+                <igx-time-picker [overlaySettings]="myOverlaySettings" format="HH:mm" mode="dropdown">
+                    <ng-template igxTimePickerTemplate let-openDialog="openDialog" let-value="value" let-displayTime="displayTime">
```

Comment:
For some reason the dropdown opens bellow the input and goes out of the screen. This is applicable to both samples.

Reply:
AutoPositionStrategy can be used instead of ConnectedPositioningStrategy for the drop-down positionStrategy.

Reply:
The `container` of the time picker has `hidden` set to true in the markup [here](https://github.com/IgniteUI/igniteui-angular/blob/3660f146cb956636aa79fe26cb1cbce223ab27df/projects/igniteui-angular/src/lib/time-picker/time-picker.component.html#L27)
When the element is hidden in the DOM it has no size. As a result you are passing sizeless element to the `overlay`. `Overlay` checks if element size goes out of the screen when auto position strategy is in use. In your case element has no size so `overlay` decides it should not fit the element in the view port.
To solve this you should either remove `hidden` and hide element some other way, or you may use `igxToggle` directive where this is solved.
Note that you are also incorrectly setting `collapsed` to `false` in [openDialog](https://github.com/IgniteUI/igniteui-angular/blob/3660f146cb956636aa79fe26cb1cbce223ab27df/projects/igniteui-angular/src/lib/time-picker/time-picker.component.ts#L1160). This should be set in `onOpened` event of the `overlay` as you are doing it [here](https://github.com/IgniteUI/igniteui-angular/blob/3660f146cb956636aa79fe26cb1cbce223ab27df/projects/igniteui-angular/src/lib/time-picker/time-picker.component.ts#L692).